packages/fxa-auth-server/lib/devices.js

```diff
@@ -89,6 +89,11 @@ module.exports = (log, db, push) => {
       if (! deviceInfo.name) {
         deviceInfo.name = credentials.client && credentials.client.name || '';
       }
+      if (credentials.client && ! deviceInfo.type) {
```

Comment:
I'm assuming `credentials.client` is only set for OAuth registrations? It's not really clear how that gets set. If so, can you update the comment to that effect?